### PR TITLE
textpieces: init at 2.3.3

### PR DESCRIPTION
**Pull request by brecert**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/textpieces/default.nix [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+, wrapGAppsHook4
+, glib
+, gtk4
+, json-glib
+, libadwaita
+, libgee
+, wayland
+, nix-update-script
+}:
+let pythonEnv = python39.withPackages (p: [ p.pyyaml ]); in
```
Comment (winterqt) [inline pkgs/applications/misc/textpieces/default.nix:23]: Nit: move the `pythonEnv` deceleration to a separate line.

**Review by winterqt — APPROVED** [review]
Other than the suggested formatting change, LGTM.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9883,6 +9883,8 @@ with pkgs;
 
   textadept11 = callPackage ../applications/editors/textadept/11 { };
 
+  textpieces = callPackage ../applications/misc/textpieces { };
```
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:9886]:
```suggestion
textpieces = callPackage ../applications/misc/textpieces { };
```

**Review by legendofmiracles — COMMENTED** [review]
Does this really only run on Linux?

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (brecert) [reply pkgs/top-level/all-packages.nix:9886]: Is the suggestion here to remove the indentation intentional?

**Comment by brecert** [comment]
> Does this really only run on Linux?

I'm not entire sure.
I looked at other GTK4 apps in nixpkgs and many/most of them are limited to linux, and the build process requires wayland libraries as well which leads me believe it can only run on linux.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (legendofmiracles) [reply pkgs/top-level/all-packages.nix:9886]: Oh, the github mobile app must have broken formatting, or I was just blind.

File: pkgs/applications/misc/textpieces/default.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+, glib
+, gtk4
+, json-glib
+, libadwaita
+, libgee
+, wayland
+, nix-update-script
+}:
+let
+  pythonEnv = python39.withPackages (p: [ p.pyyaml ]);
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/textpieces/default.nix:24]:
```suggestion
  pythonEnv = python3.withPackages (p: [ p.pyyaml ]);
```

File: pkgs/applications/misc/textpieces/default.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    wayland
+  ];
+
+  runtimeDependencies = [
+    pythonEnv
+  ];
+
+  postPatch = ''
+    patchShebangs build-aux/meson/
+    patchShebangs scripts/
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/textpieces/default.nix:66]:
```suggestion
    patchShebangs build-aux/meson/ scripts/
```

File: pkgs/applications/misc/textpieces/default.nix (hunk 3 above)
Comment (brecert) [reply pkgs/applications/misc/textpieces/default.nix:24]: May I ask what your reasoning for changing python39 to python3 is?

Comment (SuperSandro2000) [reply pkgs/applications/misc/textpieces/default.nix:24]: Right now python3 points to python39 and in the future if we change the default python version we do not want to cleanup explicit pints unless necessary.

**Pull request closed without merging**